plugin/manager.go

```diff
@@ -173,9 +173,22 @@ func (pm *Manager) reload() error { // todo: restore
 		if validFullID.MatchString(v.Name()) {
 			p, err := pm.loadPlugin(v.Name())
 			if err != nil {
-				return err
+				if os.IsNotExist(errors.Cause(err)) {
+					// Likely some error while removing on an older version of docker
+					logrus.WithError(err).WithField("id", filepath.Base(v.Name())).Warn("Missing plugin config, skipping. This may be caused due to a failed remove and requires manual cleanup.")
+				} else {
+					logrus.WithError(err).WithField("id", filepath.Base(v.Name())).Error("Error loading plugin, skipping.")
+				}
+				continue
 			}
 			plugins[p.GetID()] = p
+		} else {
```

Comment:
Perhaps we should refactor this loop a bit; there's a lot of nested conditions now. Perhaps do a `continue` in each `if` branch to reduce nesting with one level at least